lib/inference/open_ai_image_generator.rb

```diff
@@ -45,6 +288,185 @@ def self.perform!(prompt, model: "dall-e-3", size: "1024x1024", api_key: nil, ap
               raise "OpenAI API returned #{response.code} #{response.body}"
             else
               json = JSON.parse(response.body, symbolize_names: true)
+              # Add original prompt to response to preserve it
+              json[:original_prompt] = original_prompt
+            end
+          end
+          json
+        end
+      end
+
+      def self.perform_edit_api_call!(
+        images,
+        prompt,
+        mask: nil,
+        model: "gpt-image-1",
+        size: "auto",
+        api_key:,
+        api_url:,
+        n: 1,
+        quality: nil
+      )
+        uri = URI(api_url)
+
+        # Setup for multipart/form-data request
+        boundary = SecureRandom.hex
+        headers = { "Content-Type" => "multipart/form-data; boundary=#{boundary}" }
+
+        if uri.host.include?("azure")
+          headers["api-key"] = api_key
+        else
+          headers["Authorization"] = "Bearer #{api_key}"
+        end
+
+        # Create multipart form data
+        body = []
+
+        # Add model
+        body << "--#{boundary}\r\n"
+        body << "Content-Disposition: form-data; name=\"model\"\r\n\r\n"
+
+        body << "#{model}\r\n"
+
+        # Add images
+        images.each do |image|
+          image_data = nil
+          image_filename = nil
+
+          # Handle different image input types
+          if image.is_a?(Upload)
+            image_path =
+              if image.local?
+                Discourse.store.path_for(image)
+              else
+                Discourse.store.download_safe(image, max_file_size_kb: MAX_IMAGE_SIZE)&.path
+              end
+            image_data = File.read(image_path)
+            image_filename = File.basename(image.url)
+          elsif image.is_a?(File) || image.is_a?(Tempfile)
+            image_data = File.read(image.path)
+            image_filename = File.basename(image.path)
+          elsif image.is_a?(String) && File.exist?(image)
+            image_data = File.read(image)
+            image_filename = File.basename(image)
+          elsif image.is_a?(String) && image.start_with?("http")
+            # Download image from URL
+            image_temp =
```

Comment:
Same as https://github.com/discourse/discourse-ai/pull/1293/files#r2065599634

Reply:
looking at core, not seeing us doing any cleanup when we download safe?

Reply:
medium straight forward cause we need extra tracking to do that, I don't mind nuking it early, but it complicates the code

Reply:
ok I cleaned it up in my last commit